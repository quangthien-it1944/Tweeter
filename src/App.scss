// reset css
@import "/node_modules/normalize.css/normalize.css";
.rounded-0 {
    border-radius: 0% !important;
}

.d-flex {
    display: flex !important;
}

.align-items-center {
    align-items: center;
}

.align-items-end {
    align-items: flex-end;
}

.justify-content-center {
    justify-content: center;
}

.justify-content-end {
    justify-content: end;
}

.w-100 {
    width: 100% !important;
}

.jusity-content-between {
    justify-content: space-between;
}

.d-inline-block {
    display: inline-block !important;
}

#message-container {
    display: block;
    margin: 30px;
    border-top: 1px solid #333;
    border-right: 1px solid #333;
    border-left: 1px solid #333;
}

.pointer {
    cursor: pointer;
}