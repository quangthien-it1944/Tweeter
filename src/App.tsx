import React, { PureComponent } from 'react';
import AddMessageContainer from './components/Form';
import MessageContainer from "./components/Messages";
import "./App.scss";

class App extends PureComponent {

  UNSAFE_componentWillMount() {
    const font = document.createElement("link");
    font.href = "https://fonts.googleapis.com/css?family=Roboto:300,400,500";
    font.rel = "stylesheet";
    font.id = "font-roboto";

    document.getElementsByTagName('head')[0].appendChild(font);
  }

  componentWillUnmount() {
    const font: any = document.getElementById("font-roboto");
    font.parentNode.removeChild(font);
  }

  render() {
    return (
      <div className="page">
        <ul>
          <li>1. Có thể thêm sửa xóa tin nhắn</li>
          <li>2. Lỗi có thể xảy ra, cái này là random để xem trường hợp</li>
        </ul>
        <div id="message-container">
          <MessageContainer />
          <AddMessageContainer />
        </div>
      </div>
    );
  }
}

export default App;
