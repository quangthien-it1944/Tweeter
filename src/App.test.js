import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import splitMessage from "./libs/Split";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render( < App / > , div);
    ReactDOM.unmountComponentAtNode(div);
});

it("caculation string split elder 50", () => {
    const source = splitMessage("I can't believe Tweeter now supports chunking my messages, so I don't have to do it myself.");
    expect(source).toEqual(
        ["1/2 I can't believe Tweeter now supports chunking", "2/2 my messages, so I don't have to do it myself."]
    );
});