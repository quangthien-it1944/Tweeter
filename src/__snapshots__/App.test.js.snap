// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`caculation string split elder 50 1`] = `"[\\"1/2 I can't believe Tweeter now supports chunking\\",\\"2/2 my messages, so I don't have to do it myself.\\"]"`;
